src/server/pfs/server/api_server.go

```diff
@@ -101,6 +97,7 @@ func (a *apiServer) CreateRepoInTransaction(txnCtx *txncontext.TransactionContex
 
 // CreateRepo implements the protobuf pfs.CreateRepo RPC
 func (a *apiServer) CreateRepo(ctx context.Context, request *pfs.CreateRepoRequest) (response *types.Empty, retErr error) {
+	pfs.EnsureRepoProject(request.Repo)
```

Comment:
This is necessary because older clients will always send a `Repo` message with a missing `Project` member.

The idea of an interceptor is an interesting idea. I think that it should be a new story, though: CORE-1175.